src/animation/AnimationUtils.js

```diff
@@ -158,6 +158,50 @@ var AnimationUtils = {
 
 		}
 
+	},
+
+	clone: function ( source ) {
+
+		var sourceNodes = {};
+		var clonedNodes = {};
+
+		var clone = source.clone();
+
+		source.traverse( function ( node ) {
+
+			sourceNodes[ node.name ] = node;
```

Comment:
Do you assume node has an unique name?

Reply:
Unique within the subtree being cloned, yes.